### PR TITLE
Rpg playerclass/1306 incorporate speed depending on npc player class into npc indefinite movement in auto gen movement

### LINKED ISSUE #1306: Incorporate speed depending on NPC Player Class into NPC indefinite movement in auto_gen_movement()
NPC indefinite movement paths should be dependent on speed attributes associated with their specific class. Right now the time the NPC spends in each room is completely random, so the min and max times should change based on the class. We will determine how to access playerclass stats through the hash table and come up with a viable strategy to convert the speed stat into a multiplier that affects the minimum and maximum times NPCs on an indefinite movement path spend in a room.

### PR DESCRIPTION
**Pull request by Vincent-Zhang4135**
Jonathan and I finally got a successful build after trying to incorporate speed stat multiplier into auto_gen_movement() in npc_move.c. 

We used this stats hash table function found in stats.h in game-state to change the speed multipliers, and implemented this into auto_gen_movement(). However, because the Test case for Test(npc_mov, auto_gen_movement_indefinite was not written for this change, it crashed. We decided that this can be a separate task of adjusting the test to work for our new implementation/feature, and will close this issue now.

### CODE REVIEW

File: src/npc/src/dialogue.c [hunk 1]
```diff
@@ -270,7 +283,8 @@ char *create_return_string(node_t *n, int is_leaf)
         }
         cur_edge = cur_edge->next;
     }
-    if (!is_leaf) {
+    if (!is_leaf)
```
Comment (katie-gao) [inline src/npc/src/dialogue.c:286]: This is a small note, but I would try keeping the if statements consistent with syntax. Earlier in the code, there is (___ != NULL) for while and if statements, and here there is just !. I would try sticking with one for consistency.

File: src/npc/src/npc_battle.c [hunk 2]
```diff
@@ -19,21 +19,21 @@ int npc_battle_init(npc_battle_t *npc_battle, int health, stat_t* stats,
...
+    npc_battle->moves = malloc(sizeof(move_t));
 
-    int check = npc_battle_init(npc_battle, health, stats, moves, ai, 
+    int check = npc_battle_init(npc_battle, health, stats, moves, ai,
                                 hostility_level, surrender_level);
 
-    if (npc_battle == NULL || npc_battle->stats == NULL ||  
-        npc_battle->moves == NULL || check != SUCCESS)
+    if (npc_battle == NULL || npc_battle->stats == NULL ||
+            npc_battle->moves == NULL || check != SUCCESS)
```
Comment (katie-gao) [inline src/npc/src/npc_battle.c:36]: Nice job in formatting for this file!

File: src/npc/src/npc_move.c [hunk 3]
```diff
@@ -308,30 +312,35 @@ int auto_gen_movement(npc_mov_t *npc_mov, game_t *game)
...
+        if(npc_mov->mov_type == NPC_MOV_DEFINITE)
+        {
+            rc = extend_path_definite(npc_mov, room_to_add);
         }
-        else if(npc_mov->mov_type == NPC_MOV_INDEFINITE) {
+        else if(npc_mov->mov_type == NPC_MOV_INDEFINITE)
+        {
             int mintime_in_room = 30000; // min time in room in ms, 30000 ms = 30 s
             int maxtime_in_room = 90000; // max time in room in ms, 90000 ms = 90 s
             int time_in_room = (rand() % (maxtime_in_room - mintime_in_room + 1)) + mintime_in_room;
```
Comment (katie-gao) [inline src/npc/src/npc_move.c:422]: Can you guys check if this line surpasses the 80 character mark?

File: src/npc/src/npc_move.c [hunk 4]
```diff
@@ -308,30 +312,35 @@ int auto_gen_movement(npc_mov_t *npc_mov, game_t *game)
...
-        if(npc_mov->mov_type == NPC_MOV_DEFINITE) {
-        	rc = extend_path_definite(npc_mov, room_to_add);
+        if(npc_mov->mov_type == NPC_MOV_DEFINITE)
+        {
+            rc = extend_path_definite(npc_mov, room_to_add);
         }
-        else if(npc_mov->mov_type == NPC_MOV_INDEFINITE) {
+        else if(npc_mov->mov_type == NPC_MOV_INDEFINITE)
+        {
             int mintime_in_room = 30000; // min time in room in ms, 30000 ms = 30 s
```
Comment (katie-gao) [inline src/npc/src/npc_move.c:420]: This was a really clever way to incorporate the time

File: src/npc/src/npc_move.c [hunk 5]
```diff
@@ -308,30 +312,35 @@ int auto_gen_movement(npc_mov_t *npc_mov, game_t *game)
...
-	      }
+        }
 
-        if(rc == FAILURE) {
-          return rc;
+        if(rc == FAILURE)
+        {
+            return rc;
         }
     }
```
Comment (katie-gao) [inline src/npc/src/npc_move.c:431]: Overall this function looks really good and it's really cool you guys successfully learned the hashtable stuff. From my understanding, the logic makes sense and is contingent on other functions. I really liked that all the functions are short and there is good usage of helper functions for auto_gen_movement. Moving forward, I think once there is testing involved then we can actually see if everything is running as wanted. Also, the formatting and syntax of this file looks really good!

**Comment by lilyehsani** [comment]
This PR was opened on the wrong branch, we are using #1458 instead now. Closing this.

**Pull request closed without merging**